lms/djangoapps/course_home_api/outline/v1/views.py

```diff
@@ -392,17 +411,40 @@ def dismiss_welcome_message(request):
 def save_course_goal(request):
     course_id = request.data.get('course_id', None)
     goal_key = request.data.get('goal_key', None)
+    number_of_days_with_visits_per_week_goal = request.data.get('number_of_days_with_visits_per_week_goal', None)
+    subscribed_to_goal_reminders = request.data.get('subscribed_to_goal_reminders', None)
 
     # If body doesn't contain 'course_id', return 400 to client.
     if not course_id:
         raise ParseError(_("'course_id' is required."))
 
-    # If body doesn't contain 'goal', return 400 to client.
-    if not goal_key:
-        raise ParseError(_("'goal_key' is required."))
+    if COURSE_GOALS_NUMBER_OF_DAYS_GOALS.is_enabled():
+        # If body doesn't contain the required goals fields, return 400 to client.
+        if number_of_days_with_visits_per_week_goal is None or subscribed_to_goal_reminders is None:
+            raise ParseError(_("'number_of_days_with_visits_per_week_goal' and 'subscribed_to_goal_reminders' are required."))
```

Comment:
Does this get shown to the user? I'd probably skip translation if not. It's a pretty developer-oriented message anyway. User isn't going to be able to fix whatever this is. Same below.